src/codegen_fn.cpp

```
@@ -1,9 +1,40 @@
+std::string CodeGen::deriveNativePackage(const SourceLocation &loc) const {
+    if (!sm_ || loc.file_id < 0 || loc.file_id >= sm_->getFileCount())
+        return "";
+    const std::string &fname = sm_->getFilename(loc.file_id);
+
+    namespace fs = std::filesystem;
+    fs::path p(fname);
+    fs::path stem = p.stem();
+    fs::path parent = p.parent_path();
+
+    // /std/<pkg>/<pkg>.ry (directory package)
+    if (parent.filename() == stem && parent.parent_path().filename() == "std")
```

Cppcheck (2.20.0)

[style] The function 'getWarnings' is never used.

(unusedFunction)